shared/src/api/extension/flatExtentionAPI.ts

```diff
@@ -10,13 +10,26 @@ import { FlatExtHostAPI, MainThreadAPI } from '../contract'
  */
 export interface ExtState {
     settings?: Readonly<SettingsCascade<object>>
+
+    roots: readonly sourcegraph.WorkspaceRoot[]
 }
 
 export interface InitResult {
     configuration: sourcegraph.ConfigurationService
+    workspace: PartialWorkspaceNamespace
     exposedToMain: FlatExtHostAPI
 }
 
+/**
+ * mimics sourcegraph.workspace namespace without documents
+ */
+export interface PartialWorkspaceNamespace {
+    roots: readonly sourcegraph.WorkspaceRoot[]
+    onDidChangeRoots: sourcegraph.Subscribable<void>
+    rootChanges: sourcegraph.Subscribable<void>
+    versionContext: string | undefined
+    versionContextChanges: sourcegraph.Subscribable<string | undefined>
+}
```

Comment:
```suggestion
export type PartialWorkspaceNamespace = Omit<
        typeof sourcegraph['workspace'],
        'textDocuments' | 'onDidOpenTextDocument' | 'openedTextDocuments'
    >
```

This would allow exhaustively re-declaring all properties.

Reply:
nice! tnx <3 I was looking how to do it but TS had issues with namepsaces.